NPE while running tests but not on macOS

**Citrus Version**
4.8.0

**Expected behavior**
Expect no NPE on [this line](https://github.com/citrusframework/citrus/blob/v4.8.0/endpoints/citrus-camel/src/main/java/org/citrusframework/camel/actions/CamelVerifyIntegrationAction.java#L175).

**Actual behavior**
Getting NPE on `Windows 11 Enterprise 23H2` and `SUSE Linux Enterprise Server 15 SP6` as the following (however on `macOS 15.6.1 or 14.7.7` all the tests pass without any error):

```
Tests finished after 18527 ms
Total tests run: 1, Passes: 0, Failures: 1, Skips: 0

Failures (1):
  JUnit Jupiter:JUnitCitrusTest:execute():test-manual-upload-server.yaml
    MethodSource [className = 'org.citrusframework.junit.jupiter.main.JUnitCitrusTest', methodName = 'execute', methodParameterTypes = '']
    => org.citrusframework.exceptions.TestCaseFailedException: Cannot invoke "String.equals(Object)" because the return value of "java.util.Map.get(Object)" is null
       org.citrusframework.DefaultTestCase.executeAction(DefaultTestCase.java:240)
       org.citrusframework.DefaultTestCase.executeTest(DefaultTestCase.java:143)
       org.citrusframework.DefaultTestCase.doExecute(DefaultTestCase.java:124)
       org.citrusframework.actions.AbstractTestAction.execute(AbstractTestAction.java:58)
       org.citrusframework.Citrus.run(Citrus.java:126)
       [...]
     Caused by: java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "java.util.Map.get(Object)" is null
       org.citrusframework.camel.actions.CamelVerifyIntegrationAction.findProcessAndVerifyStatus(CamelVerifyIntegrationAction.java:175)
       org.citrusframework.camel.actions.CamelVerifyIntegrationAction.verifyRouteStatus(CamelVerifyIntegrationAction.java:129)
       org.citrusframework.camel.actions.CamelVerifyIntegrationAction.doExecute(CamelVerifyIntegrationAction.java:76)
       org.citrusframework.actions.AbstractTestAction.execute(AbstractTestAction.java:58)
       org.citrusframework.camel.actions.CamelRunIntegrationAction.doExecute(CamelRunIntegrationAction.java:151)
       [...]
```

**Test case sample**
I have to look into how I can provide test reproducing the issue but basically I have got a Camel Route in YAML-DSL which exposes a [HTTP Server](https://camel.apache.org/components/latest/platform-http-component.html) and in my tests I assert on it's behaviour (positive as well as negative) [after starting it as the first step](https://camel.apache.org/manual/camel-jbang-test.html#_run_the_camel_integration), something like:

```
  - http: # uploading csv should succeed
      client: ${manual-upload-url}
      sendRequest:
        POST:
          headers:
            - name: Authorization
              value: Basic bW9kZWwyNzpjaGFuZ2VpdA==
            - name: Content-Type
              value: multipart/form-data; boundary=------------------------Vp4YEeSwtKdox0IKwW7MJ9
          body:
            data: |
              --------------------------Vp4YEeSwtKdox0IKwW7MJ9
              Content-Disposition: form-data; name=test.csv; filename=test.csv
              Content-Type: application/octet-stream
              
              foo,bar,baz
              --------------------------Vp4YEeSwtKdox0IKwW7MJ9--
  - http:
      client: ${manual-upload-url}
      receiveResponse:
        response:
          status: 200
          reasonPhrase: OK
          body:
            data: "'test.csv' of the type 'text/csv' and size '11' bytes uploaded successfully.\n"
```

The same error is reproducible on Windows using the following official example where Camel Platform HTTP consumer is in use as in my case:

https://github.com/apache/camel-jbang-examples/blob/main/aws/aws-sqs/README.adoc#integration-testing

```
2025-09-16 10:00:54.323  INFO 41072 --- [           main] rk.junit.jupiter.JUnitJupiterEngine : Adding test source http-to-aws-sqs.camel.it.yaml
2025-09-16 10:00:55.350  INFO 41072 --- [           main] rusframework.report.LoggingReporter :        .__  __
2025-09-16 10:00:55.350  INFO 41072 --- [           main] rusframework.report.LoggingReporter :   ____ |__|/  |________ __ __  ______
2025-09-16 10:00:55.351  INFO 41072 --- [           main] rusframework.report.LoggingReporter : _/ ___\|  \   __\_  __ \  |  \/  ___/
2025-09-16 10:00:55.351  INFO 41072 --- [           main] rusframework.report.LoggingReporter : \  \___|  ||  |  |  | \/  |  /\___ \
2025-09-16 10:00:55.351  INFO 41072 --- [           main] rusframework.report.LoggingReporter :  \___  >__||__|  |__|  |____//____  >
2025-09-16 10:00:55.351  INFO 41072 --- [           main] rusframework.report.LoggingReporter :      \/                           \/
2025-09-16 10:00:55.351  INFO 41072 --- [           main] rusframework.report.LoggingReporter :
2025-09-16 10:00:55.352  INFO 41072 --- [           main] rusframework.report.LoggingReporter : C I T R U S  T E S T S  4.8.0
2025-09-16 10:00:55.352  INFO 41072 --- [           main] rusframework.report.LoggingReporter :
2025-09-16 10:00:55.353  INFO 41072 --- [           main] rusframework.report.LoggingReporter : ------------------------------------------------------------------------
2025-09-16 10:00:55.727  INFO 41072 --- [           main] rg.testcontainers.images.PullPolicy : Image pull policy will be performed by: DefaultPullPolicy()
2025-09-16 10:00:55.730  INFO 41072 --- [           main] ainers.utility.ImageNameSubstitutor : Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2025-09-16 10:00:55.751  INFO 41072 --- [           main] ntainers.aws2.StartLocalStackAction : Starting Testcontainers container 'aws2Container'
2025-09-16 10:00:55.752  INFO 41072 --- [           main] .testcontainers.DockerClientFactory : Testcontainers version: 1.21.3
2025-09-16 10:00:55.952  INFO 41072 --- [           main] client.DockerClientProviderStrategy : Loaded org.testcontainers.dockerclient.NpipeSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2025-09-16 10:00:56.445  INFO 41072 --- [           main] client.DockerClientProviderStrategy : Found Docker environment with local Npipe socket (npipe:////./pipe/docker_engine)
2025-09-16 10:00:56.449  INFO 41072 --- [           main] .testcontainers.DockerClientFactory : Docker host IP address is localhost
2025-09-16 10:00:56.470  INFO 41072 --- [           main] .testcontainers.DockerClientFactory : Connected to docker:
  Server Version: 28.1.1
  API Version: 1.49
  Operating System: Docker Desktop
  Total Memory: 31871 MB
  Labels:
    com.docker.desktop.address=npipe://\\.\pipe\docker_cli
2025-09-16 10:00:56.497  INFO 41072 --- [           main] tc.testcontainers/ryuk:0.12.0       : Creating container for image: testcontainers/ryuk:0.12.0
2025-09-16 10:00:57.143  INFO 41072 --- [           main] tainers.utility.RegistryAuthLocator : Credential helper/store (docker-credential-desktop) does not have credentials for https://index.docker.io/v1/
2025-09-16 10:00:57.340  INFO 41072 --- [           main] tc.testcontainers/ryuk:0.12.0       : Container testcontainers/ryuk:0.12.0 is starting: c6ffb0652948ed40fd80588578ed52d0e0b6246a242dfa3f109a435c1dbcb64f
2025-09-16 10:00:57.749  INFO 41072 --- [           main] tc.testcontainers/ryuk:0.12.0       : Container testcontainers/ryuk:0.12.0 started in PT1.2521228S
2025-09-16 10:00:57.758  INFO 41072 --- [           main] ntainers.utility.RyukResourceReaper : Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
2025-09-16 10:00:57.758  INFO 41072 --- [           main] .testcontainers.DockerClientFactory : Checking the system...
2025-09-16 10:00:57.759  INFO 41072 --- [           main] .testcontainers.DockerClientFactory : Γ£ö∩╕Ä Docker server version should be at least 1.6.0
2025-09-16 10:00:57.760  INFO 41072 --- [           main] tc.localstack/localstack:4.2.0      : HOSTNAME_EXTERNAL environment variable set to localhost (to match host-routable address for container)
2025-09-16 10:00:57.761  INFO 41072 --- [           main] tc.localstack/localstack:4.2.0      : Creating container for image: localstack/localstack:4.2.0
2025-09-16 10:00:58.113  INFO 41072 --- [           main] tc.localstack/localstack:4.2.0      : Container localstack/localstack:4.2.0 is starting: 5ba0b04c3a42188a69e7390e9e32e60bf64a5d040e13b071ee24212748f7ae33
2025-09-16 10:01:00.432  INFO 41072 --- [           main] tc.localstack/localstack:4.2.0      : Container localstack/localstack:4.2.0 started in PT2.6711808S
2025-09-16 10:01:01.009  INFO 41072 --- [           main] ntainers.aws2.StartLocalStackAction : Successfully started Testcontainers container 'aws2Container (competent_jemison)' with id 5ba0b04c3a42188a69e7390e9e32e60bf64a5d040e13b071ee24212748f7ae33
2025-09-16 10:01:01.303  INFO 41072 --- [           main] el.impl.engine.AbstractCamelContext : Apache Camel 4.14.0 (camel-1) is starting
2025-09-16 10:01:01.318  INFO 41072 --- [           main] el.impl.engine.AbstractCamelContext : Routes startup (total:0)
2025-09-16 10:01:01.318  INFO 41072 --- [           main] el.impl.engine.AbstractCamelContext : Apache Camel 4.14.0 (camel-1) started in 13ms (build:0ms init:0ms start:13ms)
2025-09-16 10:01:01.322  INFO 41072 --- [           main] mel.actions.StartCamelContextAction : Started Camel context 'camelContext'
2025-09-16 10:01:14.029  INFO 41072 --- [           main] l.actions.CamelRunIntegrationAction : Starting Camel integration 'http-to-aws-sqs' ...
2025-09-16 10:01:14.061  INFO 41072 --- [           main] l.actions.CamelRunIntegrationAction : Started Camel integration 'http-to-aws-sqs' (9596)
2025-09-16 10:01:14.066  INFO 41072 --- [           main] l.actions.CamelRunIntegrationAction : Waiting for the Camel integration 'http-to-aws-sqs' (9596) to be running ...
2025-09-16 10:01:14.485  INFO 41072 --- [           main] ctions.CamelVerifyIntegrationAction : Verify Camel integration 'http-to-aws-sqs' ...
2025-09-16 10:01:14.486  INFO 41072 --- [           main] INTEGRATION_STATUS                  : Waiting for Camel integration 'http-to-aws-sqs' to be in state 'Running'
2025-09-16 10:02:10.916  INFO 41072 --- [           main] ctions.CamelVerifyIntegrationAction :
  PID   NAME             READY  STATUS   AGE  TOTAL  FAIL  INFLIGHT
 12004  http-to-aws-sqs   1/1   Running  54s      0     0         0

2025-09-16 10:02:10.917  INFO 41072 --- [           main] ctions.CamelVerifyIntegrationAction : Waiting for Camel integration 'http-to-aws-sqs' to be in state 'Running'- retry in 2000 ms
2025-09-16 10:02:27.849  INFO 41072 --- [           main] rs.actions.StopTestcontainersAction : Successfully stopped Testcontainers container ' (competent_jemison)' with id 5ba0b04c3a42188a69e7390e9e32e60bf64a5d040e13b071ee24212748f7ae33
2025-09-16 10:02:32.288  INFO 41072 --- [           main] .actions.CamelStopIntegrationAction : Stopping Camel integration 'http-to-aws-sqs' ...
2025-09-16 10:02:42.461  INFO 41072 --- [           main] .actions.CamelStopIntegrationAction : Stopped Camel integration 'http-to-aws-sqs' (9596 - 14036)
2025-09-16 10:02:47.256  INFO 41072 --- [           main] .actions.CamelStopIntegrationAction : Stopped Camel integration 'http-to-aws-sqs' (9596 - 12004)
2025-09-16 10:02:47.257  INFO 41072 --- [           main] .actions.CamelStopIntegrationAction : Stopped Camel integration 'http-to-aws-sqs' (9596)
2025-09-16 10:02:47.257  INFO 41072 --- [           main] rusframework.report.LoggingReporter :
2025-09-16 10:02:47.257 ERROR 41072 --- [           main] rusframework.report.LoggingReporter : TEST FAILED http-to-aws-sqs <org.citrusframework> Nested exception is: Cannot invoke "String.equals(Object)" because the return value of "java.util.Map.get(Object)" is null
2025-09-16 10:02:47.258  INFO 41072 --- [           main] rusframework.report.LoggingReporter : ------------------------------------------------------------------------
2025-09-16 10:02:47.258  INFO 41072 --- [           main] rusframework.report.LoggingReporter :
2025-09-16 10:02:47.272  INFO 41072 --- [           main] rusframework.report.LoggingReporter :
2025-09-16 10:02:47.272  INFO 41072 --- [           main] rusframework.report.LoggingReporter : CITRUS TEST RESULTS
2025-09-16 10:02:47.273  INFO 41072 --- [           main] rusframework.report.LoggingReporter :
2025-09-16 10:02:47.273  INFO 41072 --- [           main] rusframework.report.LoggingReporter : FAILURE ( 91801 ms) http-to-aws-sqs
2025-09-16 10:02:47.276  INFO 41072 --- [           main] rusframework.report.LoggingReporter :         Caused by: NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "java.util.Map.get(Object)" is null
2025-09-16 10:02:47.276  INFO 41072 --- [           main] rusframework.report.LoggingReporter :
2025-09-16 10:02:47.277  INFO 41072 --- [           main] rusframework.report.LoggingReporter : TOTAL:          1
2025-09-16 10:02:47.277  INFO 41072 --- [           main] rusframework.report.LoggingReporter : PASSED:         0 (0.0%)
2025-09-16 10:02:47.279  INFO 41072 --- [           main] rusframework.report.LoggingReporter : FAILED:         1 (100.0%)
2025-09-16 10:02:47.280  INFO 41072 --- [           main] rusframework.report.LoggingReporter : TIME:           91801 ms
2025-09-16 10:02:47.281  INFO 41072 --- [           main] rusframework.report.LoggingReporter :
2025-09-16 10:02:47.281  INFO 41072 --- [           main] rusframework.report.LoggingReporter : ------------------------------------------------------------------------
2025-09-16 10:02:47.296  INFO 41072 --- [           main] k.report.AbstractOutputFileReporter : Generated test report: .citrus-jbang\citrus-reports\citrus-test-results.html

Tests finished after 112681 ms
Total tests run: 1, Passes: 0, Failures: 1, Skips: 0

Failures (1):
  JUnit Jupiter:JUnitCitrusTest:execute():http-to-aws-sqs.camel.it.yaml
    MethodSource [className = 'org.citrusframework.junit.jupiter.main.JUnitCitrusTest', methodName = 'execute', methodParameterTypes = '']
    => org.citrusframework.exceptions.TestCaseFailedException: Cannot invoke "String.equals(Object)" because the return value of "java.util.Map.get(Object)" is null
       org.citrusframework.DefaultTestCase.executeAction(DefaultTestCase.java:240)
       org.citrusframework.DefaultTestCase.executeTest(DefaultTestCase.java:143)
       org.citrusframework.DefaultTestCase.doExecute(DefaultTestCase.java:124)
       org.citrusframework.actions.AbstractTestAction.execute(AbstractTestAction.java:58)
       org.citrusframework.Citrus.run(Citrus.java:126)
       [...]
     Caused by: java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "java.util.Map.get(Object)" is null
       org.citrusframework.camel.actions.CamelVerifyIntegrationAction.findProcessAndVerifyStatus(CamelVerifyIntegrationAction.java:175)
       org.citrusframework.camel.actions.CamelVerifyIntegrationAction.verifyRouteStatus(CamelVerifyIntegrationAction.java:147)
       org.citrusframework.camel.actions.CamelVerifyIntegrationAction.doExecute(CamelVerifyIntegrationAction.java:76)
       org.citrusframework.actions.AbstractTestAction.execute(AbstractTestAction.java:58)
       org.citrusframework.camel.actions.CamelRunIntegrationAction.doExecute(CamelRunIntegrationAction.java:151)
       [...]
```

--- a/endpoints/citrus-camel/src/main/java/org/citrusframework/camel/jbang/CamelJBang.java
+++ b/endpoints/citrus-camel/src/main/java/org/citrusframework/camel/jbang/CamelJBang.java
@@ -63,10 +63,7 @@ public class CamelJBang {
 
     private final ObjectMapper objectMapper = new ObjectMapper();
 
-    /**
-     * Prevent direct instantiation.
-     */
-    private CamelJBang() {
+    public CamelJBang() {
         if (!"latest".equals(CamelJBangSettings.getCamelVersion())) {
             app.withSystemProperty("camel.jbang.version", CamelJBangSettings.getCamelVersion());
         }
@@ -276,14 +273,20 @@ public class CamelJBang {
         try {
             String output = ps("--json");
             // cleans lines starting with unprintable characters (happens on Fedora)
-            output = output.replaceAll("(?m)^\\p{C}.*\\R?","");
+            output = output.replaceAll("(?m)^\\p{C}.*\\R?","").trim();
 
-            if (!output.isBlank() && MessagePayloadUtils.isJson(output)) {
-                return objectMapper.readValue(output, new TypeReference<>() {});
-            } else {
+            if (!output.startsWith("[{") && output.contains("[{")) {
+                // obviously some unexpected log output before the Json object - remove it
+                output = output.substring(output.indexOf("[{"));
+            }
+
+            if (output.isBlank() || !MessagePayloadUtils.isJson(output)) {
+                // no proper Json output
                 // fallback to extracting integration details from plaintext ascii table output (used by older Camel versions)
                 return parseAll();
             }
+
+            return objectMapper.readValue(output, new TypeReference<>() {});
         } catch (JsonProcessingException e) {
             logger.warn("Failed to list integrations from Camel JBang");
 
@@ -309,10 +312,23 @@ public class CamelJBang {
         }
 
         try (BufferedReader reader = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(output.getBytes(StandardCharsets.UTF_8))))) {
-            String line = reader.readLine();
+            final List<String> names = new ArrayList<>();
+            String line;
 
-            List<String> names = new ArrayList<>(Arrays.asList(line.trim().split("\\s+")));
+            // Parse Ascii table headers 1st
+            while ((line = reader.readLine()) != null) {
+                if (line.trim().startsWith("PID")) {
+                    names.addAll(Arrays.asList(line.trim().split("\\s+")));
+                    break;
+                }
+            }
 
+            if (names.isEmpty()) {
+               logger.warn("Failed to list integrations from Camel JBang - no proper Ascii table command output");
+               return integrations;
+            }
+
+            // Parse table rows and values
             while ((line = reader.readLine()) != null) {
                 Map<String, String> properties = new HashMap<>();
                 List<String> values = new ArrayList<>(Arrays.asList(line.trim().split("\\s+")));

--- a/endpoints/citrus-camel/src/test/java/org/citrusframework/camel/actions/CamelVerifyIntegrationActionTest.java
+++ b/endpoints/citrus-camel/src/test/java/org/citrusframework/camel/actions/CamelVerifyIntegrationActionTest.java
@@ -1,0 +1,293 @@
+/*
+ * Copyright the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.citrusframework.camel.actions;
+
+import org.citrusframework.camel.UnitTestSupport;
+import org.citrusframework.camel.jbang.CamelJBang;
+import org.citrusframework.exceptions.ActionTimeoutException;
+import org.citrusframework.jbang.ProcessAndOutput;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.when;
+
+public class CamelVerifyIntegrationActionTest extends UnitTestSupport {
+
+    @Mock
+    private ProcessAndOutput pao;
+
+    @Mock
+    private Process process;
+
+    @BeforeTest
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        when(pao.getProcess()).thenReturn(process);
+    }
+
+    @DataProvider(name = "commandOutputProvider")
+    public Object[][] commandOutputProvider() {
+        return new Object[][] {
+            {
+            """
+             PID   NAME      READY  STATUS   AGE  TOTAL  FAIL  INFLIGHT
+            12345  my-route   1/1   Running  10s      0     0         0
+            """
+            },
+            {
+            """
+            some unexpected log output
+             PID   NAME      READY  STATUS   AGE  TOTAL  FAIL  INFLIGHT
+            12345  my-route   1/1   Running  10s      0     0         0
+            """
+            },
+            {
+            """
+             PID   NAME         READY  STATUS   AGE  TOTAL  FAIL  INFLIGHT
+            11111  other-route   1/1   Running  10s      0     0         0
+            12345  my-route      1/1   Running  10s      0     0         0
+            """
+            },
+            {
+            """
+             PID   NAME         READY  STATUS   AGE  TOTAL  FAIL  INFLIGHT
+            11111  other-route   1/1   Running  10s      0     0         0
+            99999  my-route      1/1   Running  10s      0     0         0
+            """
+            }
+        };
+    }
+
+    @Test(dataProvider = "commandOutputProvider")
+    public void shouldParseOutputAndVerifyRunningStatus(String output) {
+        when(process.isAlive()).thenReturn(true);
+        when(process.exitValue()).thenReturn(0);
+
+        when(pao.getOutput()).thenReturn(output);
+
+        context.getReferenceResolver().bind("camelJBang", new CamelJBang() {
+            @Override
+            public String ps(String... args) {
+                if (args.length == 1 && args[0].equals("--json")) {
+                    // Disable Json output on purpose
+                    return "";
+                }
+                return super.ps(args);
+            }
+
+            @Override
+            public String ps() {
+                // Mock the ps() output
+                return output;
+            }
+        });
+
+        context.setVariable("my-route:pid", 12345);
+
+        CamelVerifyIntegrationAction action = new CamelVerifyIntegrationAction.Builder()
+                .integration("my-route")
+                .maxAttempts(5)
+                .delayBetweenAttempts(100)
+                .withReferenceResolver(context.getReferenceResolver())
+                .build();
+
+        action.execute(context);
+    }
+
+    @DataProvider(name = "commandJsonOutputProvider")
+    public Object[][] commandJsonOutputProvider() {
+        return new Object[][] {
+            {
+            """
+            [{"pid":12345,"name":"my-route","ready":"1/1","status":"Running","age":"25s","total":0,"fail":0,"inflight":0}]
+            """
+            },
+            {
+            """
+            some unexpected log output
+            [{"pid":12345,"name":"my-route","ready":"1/1","status":"Running","age":"25s","total":0,"fail":0,"inflight":0}]
+            """
+            },
+            {
+            """
+            [{"pid":11111,"name":"other-route","ready":"1/1","status":"Running","age":"25s","total":0,"fail":0,"inflight":0},{"pid":12345,"name":"my-route","ready":"1/1","status":"Running","age":"25s","total":0,"fail":0,"inflight":0}]
+            """
+            },
+            {
+            """
+            [{"pid":11111,"name":"other-route","ready":"1/1","status":"Running","age":"25s","total":0,"fail":0,"inflight":0},{"pid":99999,"name":"my-route","ready":"1/1","status":"Running","age":"25s","total":0,"fail":0,"inflight":0}]
+            """
+            }
+        };
+    }
+
+    @Test(dataProvider = "commandJsonOutputProvider")
+    public void shouldVerifyRunningStatus(String output) {
+        when(process.isAlive()).thenReturn(true);
+        when(process.exitValue()).thenReturn(0);
+
+        when(pao.getOutput()).thenReturn(output);
+
+        context.getReferenceResolver().bind("camelJBang", new CamelJBang() {
+            @Override
+            public String ps(String... args) {
+                if (args.length == 1 && args[0].equals("--json")) {
+                    return output;
+                }
+                return super.ps(args);
+            }
+
+            @Override
+            public String ps() {
+                return "";
+            }
+        });
+
+        context.setVariable("my-route:pid", 12345);
+
+        CamelVerifyIntegrationAction action = new CamelVerifyIntegrationAction.Builder()
+                .integration("my-route")
+                .maxAttempts(5)
+                .delayBetweenAttempts(100)
+                .withReferenceResolver(context.getReferenceResolver())
+                .build();
+
+        action.execute(context);
+    }
+
+    @DataProvider(name = "failedOutputProvider")
+    public Object[][] failedOutputProvider() {
+        return new Object[][] {
+            {""},
+            {"Something completely different"},
+            {
+            """
+             PID   NAME   READY  STATUS   AGE  TOTAL  FAIL  INFLIGHT
+            """
+            },
+            {
+            """
+             PID   NAME         READY  STATUS   AGE  TOTAL  FAIL  INFLIGHT
+            11111  other-route   1/1   Running  10s      0     0         0
+            """
+            },
+            {
+            """
+             PID   NAME      READY  STATUS   AGE  TOTAL  FAIL  INFLIGHT
+            12345  my-route   0/1   Pending  10s      0     0         0
+            """
+            },
+        };
+    }
+
+    @Test(dataProvider = "failedOutputProvider", expectedExceptions = ActionTimeoutException.class)
+    public void shouldParseOutputAndFailVerification(String output) {
+        when(process.isAlive()).thenReturn(true);
+        when(process.exitValue()).thenReturn(0);
+
+        when(pao.getOutput()).thenReturn(output);
+
+        context.getReferenceResolver().bind("camelJBang", new CamelJBang() {
+            @Override
+            public String ps(String... args) {
+                if (args.length == 1 && args[0].equals("--json")) {
+                    // Disable Json output on purpose
+                    return "";
+                }
+                return super.ps(args);
+            }
+
+            @Override
+            public String ps() {
+                // Mock the ps() output
+                return output;
+            }
+        });
+
+        context.setVariable("my-route:pid", 12345);
+
+        CamelVerifyIntegrationAction action = new CamelVerifyIntegrationAction.Builder()
+                .integration("my-route")
+                .maxAttempts(5)
+                .delayBetweenAttempts(100)
+                .withReferenceResolver(context.getReferenceResolver())
+                .build();
+
+        action.execute(context);
+    }
+
+    @DataProvider(name = "failedJsonOutputProvider")
+    public Object[][] failedJsonOutputProvider() {
+        return new Object[][] {
+            {""},
+            {"Something completely different"},
+            {
+            """
+            []
+            """
+            },
+            {
+            """
+            [{"pid":11111,"name":"other-route","ready":"1/1","status":"Running","age":"25s","total":0,"fail":0,"inflight":0}]
+            """
+            },
+            {
+            """
+            [{"pid":12345,"name":"my-route","ready":"0/1","status":"Pending","age":"25s","total":0,"fail":0,"inflight":0}]
+            """
+            }
+        };
+    }
+
+    @Test(dataProvider = "failedOutputProvider", expectedExceptions = ActionTimeoutException.class)
+    public void shouldFailVerification(String output) {
+        when(process.isAlive()).thenReturn(true);
+        when(process.exitValue()).thenReturn(0);
+
+        when(pao.getOutput()).thenReturn(output);
+
+        context.getReferenceResolver().bind("camelJBang", new CamelJBang() {
+            @Override
+            public String ps(String... args) {
+                if (args.length == 1 && args[0].equals("--json")) {
+                    return output;
+                }
+                return super.ps(args);
+            }
+
+            @Override
+            public String ps() {
+                return "";
+            }
+        });
+
+        context.setVariable("my-route:pid", 12345);
+
+        CamelVerifyIntegrationAction action = new CamelVerifyIntegrationAction.Builder()
+                .integration("my-route")
+                .maxAttempts(5)
+                .delayBetweenAttempts(100)
+                .withReferenceResolver(context.getReferenceResolver())
+                .build();
+
+        action.execute(context);
+    }
+
+}

